src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/tunnel.py

```diff
@@ -0,0 +1,192 @@
+            basic_auth_header = 'Authorization: Basic {}'.format(basic_auth_string)
+            cli_logger = get_logger()  # get CLI logger which has the level set through command lines
+            is_verbose = any(handler.level <= logs.INFO for handler in cli_logger.handlers)
+            if is_verbose:
+                logger.info('Websocket tracing enabled')
+                websocket.enableTrace(True)
+            else:
+                logger.warning('Websocket tracing disabled, use --verbose flag to enable')
+                websocket.enableTrace(False)
+            self.ws = create_connection(host,
+                                        sockopt=((socket.IPPROTO_TCP, socket.TCP_NODELAY, 1),),
+                                        class_=TunnelWebSocket,
+                                        header=[basic_auth_header],
+                                        sslopt={'cert_reqs': ssl.CERT_NONE},
+                                        enable_multithread=True)
+            logger.info('Websocket, connected status: %s', self.ws.connected)
+            index = index + 1
+            logger.info('Got debugger connection... index: %s', index)
+            debugger_thread = Thread(target=self._listen_to_client, args=(self.client, self.ws, index))
+            web_socket_thread = Thread(target=self._listen_to_web_socket, args=(self.client, self.ws, index))
+            debugger_thread.start()
+            web_socket_thread.start()
+            logger.info('Both debugger and websocket threads started...')
+            logger.warning('Successfully connected to local server..')
+            debugger_thread.join()
+            web_socket_thread.join()
+            logger.info('Both debugger and websocket threads stopped...')
+            logger.warning('Stopped local server..')
+
+    def _listen_to_web_socket(self, client, ws_socket, index):
+        while True:
+            try:
+                logger.info('Waiting for websocket data, connection status: %s, index: %s', ws_socket.connected, index)
+                data = ws_socket.recv()
+                logger.info('Received websocket data: %s, index: %s', data, index)
+                if data:
+                    # Set the response to echo back the recieved data
+                    response = data
+                    logger.info('Sending to debugger, response: %s, index: %s', response, index)
+                    client.sendall(response)
+                    logger.info('Done sending to debugger, index: %s', index)
+                else:
+                    logger.info('Client disconnected!, index: %s', index)
+                    client.close()
+                    ws_socket.close()
+                    break
+            except:
+                traceback.print_exc(file=sys.stdout)
+                client.close()
+                ws_socket.close()
+                return False
```

Comment:
Is this boolean return value used? It's best to have consistent return values, either a None, or a Bool